lib/database_consistency/checkers/index_checkers/unique_index_checker.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module DatabaseConsistency
+  module Checkers
+    # This class checks missing presence validator
+    class UniqueIndexChecker < IndexChecker
+      # Message templates
+      VALIDATOR_MISSING = 'index is unique in the database but do not have uniqueness validator'
+
+      private
+
+      # We skip check when:
+      #  - index is not unique
+      def preconditions
+        index.unique
+      end
+
+      # Table of possible statuses
+      # | validation | status |
+      # | ---------- | ------ |
+      # | provided   | ok     |
+      # | missing    | fail   |
+      #
+      def check
+        if valid?
+          report_template(:ok)
+        else
+          report_template(:fail, VALIDATOR_MISSING)
+        end
+      end
+
+      def valid?
+        model.validators.grep(ActiveRecord::Validations::UniquenessValidator).any? do |validator|
```

Comment:
I know that I have it in other places too but let's use `kind` of the validator class instead. Some libraries, like `database_validations` overwrites default validations, so they have custom classes but the same `kind`. 

WDYT?